mailpoet/assets/js/src/wizard/woocommerce_controller.jsx

```diff
@@ -71,10 +82,12 @@ function WooCommerceController({ isWizardStep = false }) {
 
 WooCommerceController.propTypes = {
   isWizardStep: PropTypes.bool,
+  redirectToNextStep: PropTypes.shape,
```

Comment:
@costasovo, one thing that I forgot to ask before is about the way that I defined the type of this parameter. Is just using `PropTypes.shape` here considered enough? Maybe there is a way to better defined what this parameter is about?

Reply:
@rodrigoprimo Good question! I missed that. Hmm, I think we used to have so lint check for that, but maybe we removed that as we are shifting toward TS. Ideally [as we discussed](https://a8c.slack.com/archives/C01GH835U90/p1667207368379109), we should try to rewrite .jsx files to Typescript when we touch them when we work on a feature. In such a case, we would remove the proptypes definitions and replace them with TS type definitions where we don't allow the `object` type.

Sometimes it is not easy to refactor to TS, so it is ok to keep the JSX file. What I wanted to say is that we will fix it during the rewrite to TS, and the component is not reused in many places (compared to some UI components), so I think we can live with the `shape`. But feel free to add a more specific definition of the property, or you can try refactoring to Typescript. Here is [a commit where I recently refactored o component on a similar level of abstraction](https://github.com/mailpoet/mailpoet/pull/4609/commits/55baa581a4433ea9acee5c9cebfc905439d88078) you may need to do similar changes if you decide to give it a try. I'm happy to assist.

Reply:
@costasovo, I completely forgot about refactoring JSX files to TSX. I added a couple of commits with my attempt in this conversation. This is one of the first times that I have done this, so there is a chance that there is something wrong. Could you please review this PR again?

Unfortunately, I had to rebase this PR and then force-push the commits, so the hash of the commits that you already reviewed changed. Also, I did not refactor steps_numbers.jsx and welcome_wizard_controller.jsx as I addressed those in #4603.

For some reason, GitHub is not picking up that woo_commerce_step.jsx was renamed to woocommerce_step.tsx and is showing woocommerce_step.tsx as a new file. My local git client is correctly identifying the rename and showing only the specific changes in the file making it easier to review.

> Here is [a commit where I recently refactored o component on a similar level of abstraction](https://github.com/mailpoet/mailpoet/pull/4609/commits/55baa581a4433ea9acee5c9cebfc905439d88078) you may need to do similar changes if you decide to give it a try.

Thanks, having an example was super helpful.